apps/s_client.c

```diff
@@ -2326,7 +2325,6 @@ int s_client_main(int argc, char **argv)
 
     read_tty = 1;
     write_tty = 0;
-    tty_on = 0;
```

Comment:
Hmm, I presume this is a dead store because `tty_on` is assigned to at lines 2920 (if) and 2922 (else), but of this and line 2920 I'd rather this line was kept as an initialisation, and the redundant assignment to `0` at line 2920 was removed - because that would then be safer in terms of what happens if code is refactored in the future (i.e. we would always end up with an initialisation).

And since this is such a large block of code, it's quite possible it will be refactored at some point in the future, and at that time I'd rather not be reliant on compiler warnings about variables used without being initialised (since compilers can and do get that wrong).

Reply:
Bah, and that wouldn't actually work, since the assignments to `tty_on` in the if-else are in for loop, so removing the `tty_on = 0` there would change behaviour.

This is an example of where the current OpenSSL coding standard of C89-style "only declare variables at the start of a function or block" is at odds with modern programming best practices of "give variables the shortest scope possible" and defensive programming and static analysis.

I would rather keep the dead assignment here to be defensive against future changes to this code where static analysis might not catch an uninitialised variable.